### PR TITLE
Rather than creating a copy-data object in build_one_patch, build it locally and move it

### PR DESCRIPTION
**Pull request by bangerth**
This avoids a bunch of rather annoying remote-NUMA-region memory write contention issues
that @kronbichler identified.

@kronbichler -- this is completely untested, and I really need to go to bed now. Is this roughly what you had in mind? If you want to time and see whether that addresses part of the problem, then I can test this correctly tomorrow and formally submit it for merge.

### CODE REVIEW

**Comment by bangerth** [comment]
Passes the testsuite.

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -1584,6 +1584,31 @@ namespace DataOutBase
...
+//      unsigned int neighbors[dim > 0
+//                             ?
+//                             GeometryInfo<dim>::faces_per_cell
+//                             :
+//                             1];
+//      unsigned int patch_index;
+//      unsigned int n_subdivisions;
+//      Table<2,float> data;
+//      bool points_are_available;
+  }
```
Comment (kronbichler) [inline source/base/data_out_base.cc]: Any reason to keep this commented-out code? It seems to be a copy of the struct members.

Comment (bangerth) [inline source/base/data_out_base.cc]: Yes, complete artifact of how I wrote this function. Absolutely no good reason to leave it in. Absolutely no good reason in fact why I should submit a patch that still has this in there :-(

In any case, I've fixed it.

File: source/numerics/data_out.cc [hunk 2]
```diff
@@ -62,31 +62,39 @@ namespace internal
...
+                 const unsigned int                                                    n_subdivisions,
+                 const unsigned int                                                    n_datasets,
+                 const CurvedCellRegion                                                curved_cell_region,
                  std::vector<DataOutBase::Patch<DH::dimension, DH::space_dimension> > &patches)
 {
+  // first create the output object that we will write into
+  ::dealii::DataOutBase::Patch<DH::dimension, DH::space_dimension> patch;
+  patch.n_subdivisions = n_subdivisions;
+  patch.data.reinit (n_datasets,
+                     Utilities::fixed_power<DH::dimension>(n_subdivisions+1));
```
Comment (kronbichler) [inline source/numerics/data_out.cc]: Why do you initialize patch.data here? Isn't line 120/131 in the new version taking care of this?

**Comment by kronbichler** [comment]
apart from the other comment, this is fine. Please adjust the code and merge yourself.

File: source/numerics/data_out.cc (hunk 2 above)
Comment (bangerth) [inline source/numerics/data_out.cc]: Good catch. Totally non-obvious too!

**Comment by bangerth** [comment]
Ah, crap, I merged this without pushing the changes @kronbichler asked for (which I wrote and tested). Will create a separate PR for this and merge it myself.

**Pull request merged**